### DISCUSSION TITLE
Would it be possible to sync calendar color?

### DISCUSSION THREAD

**Discussion by cwchristerw** [bitfireAT/icsx5#29] [Ideas]
I'm looking for solution to get color directly from X-APPLE-CALENDAR-COLOR. Currently it doesn't work for my own calendar, I'm not sure if its implemented or not.

**Comment by rfc2822** [comment]
Is there some standard or definition of `X-APPLE-CALENDAR-COLOR`? Or at least some documentation on how it is used?

**Comment by cwchristerw** [comment]
Nextcloud (using SabreDav) has its value as HEX like `#ffffff`.
https://github.com/sabre-io/dav/blob/c1afdc77a95efea6ee40c03c45f57c3c0c80ec22/lib/CalDAV/ICSExportPlugin.php#L299

Here few resources that also reference X-APPLE-CALENDAR-COLOR to be HEX color with #:
- https://github.com/apple/ccs-calendarserver/search?q=X-APPLE-CALENDAR-COLOR&type=code
- https://github.com/ics-py/ics-py/discussions/278
- https://github.com/rianjs/ical.net/issues/348
- https://github.com/gggard/AndroidCaldavSyncAdapater/issues/143
- https://github.com/sabre-io/vobject/blob/d8a0a9ae215a8acfb51afc29101c7344670b9c83/tests/VObject/EmClientTest.php
- https://github.com/sabre-io/dav/blob/c1afdc77a95efea6ee40c03c45f57c3c0c80ec22/tests/Sabre/CalDAV/ICSExportPluginTest.php

Here few resources that also reference X-APPLE-CALENDAR-COLOR can be empty:
- https://github.com/sabre-io/dav/issues/1148

X-APPLE-CALENDAR-COLOR can be HEX Color (like `#ffffff` and `#FFFFFF`) or empty

**Reply by cwchristerw** [reply]
Successor to `X-APPLE-CALENDAR-COLOR` would be `COLOR`.
https://datatracker.ietf.org/doc/html/rfc7986

This isn't implemented in any major groupware by my knowledge.

**Reply by rfc2822** [reply]
Thanks, so we could implement `COLOR` and as a fallback `X-APPLE-CALENDAR-COLOR`.

**Reply by ghost** [reply]
> Thanks, so we could implement `COLOR` and as a fallback `X-APPLE-CALENDAR-COLOR`.

I would love that new feature!

**Reply by devvv4ever** [reply]
Just sent you the apk. Happy testing!

**Reply by ghost** [reply]
Yes, thank you.
I tested it.
Here comes the result:
The `COLOR` value seems to be ignored, it's always using the `X-APPLE-CALENDAR-COLOR` hex value.
~Additionally, the list of 'Abbonierte Kalender' is always empty and the slider of 'Zugangsdaten benötigt' is resetting itself to off.... Why?~ 3 days later: problem solved, don't know why.

**Reply by rfc2822** [reply]
Can you provide that iCalendar? Which `COLOR` value does it have? `COLOR` is expected to be a CSS3 color name, `X-APPLE-CALENDAR-COLOR` is a hex value.

**Reply by ghost** [reply]
Yes. May I send the URL to your email inbox as I requested the APK?

**Reply by rfc2822** [reply]
Yes of course

**Reply by ghost** [reply]
You've got mail!

**Comment by rfc2822** [comment]
See #31

**Comment by rfc2822** [comment]
@cwchristerw This has been implemented in the latest dev version. Would you like to test? Where did you download ICSx⁵ from?

**Reply by ghost** [reply]
Sure! From F-Droid

**Reply by rfc2822** [reply]
I can send you an APK but you will have to uninstall the F-Droid version first (they use their own signing key). Where shall I send the APK too?

**Reply by ghost** [reply]
No problem. You may send me via direct message or post in this thread.

**Reply by rfc2822** [reply]
How can I send a direct message? I don't want the APK or its link to be public.

Edit: There's an email address at the top of the @bitfireAT page. You could send an email there and then I can send the APK back.

**Reply by ghost** [reply]
E-Mail sent. Thanks!

**Comment by cwchristerw** [comment]
@rfc2822 Thanks for implementing this so quickly.
